src/librustc/ty/mod.rs

```diff
@@ -2340,8 +2340,8 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
     pub fn with_freevars<T, F>(self, fid: NodeId, f: F) -> T where
         F: FnOnce(&[hir::Freevar]) -> T,
     {
-        let hir_id = self.hir.node_to_hir_id(fid);
-        match self.freevars(hir_id) {
+        let def_id = self.hir.local_def_id(fid);
+        match self.freevars(def_id) {
```

Comment:
So `with_freevars` used to be a thing because of a `RefCell`. Can it be removed in favor of `tcx.freevars(def_id)`? All callers should, ironically, already have a closure `DefId` around.